### ISSUE TITLE
racattack provisioning

### ISSUE THREAD

**Issue by pioro** [racattack/racattack-ansible-oracle#1]
Hi,

I cloned you repo today and was trying to deploy rac attack env using vagrant and ansible.

It failed twice:
1. once during db inst file unzip 

==> collabn1: failed: [collabn1] => (item=[{u'version': u'11.2.0.4', u'filename': u'p13390677_112040_Linux-x86-64_1of7.zip'}, {u'oracle_init_params': u'open_cursors=300,processes=500', u'is_racone': u'false', u'oracle_db_type': u'RAC', u'service_name': u'racdb_serv', u'storage_type': u'ASM', u'oracle_db_passwd': u'Password1', u'home': u'rachome1', u'oracle_db_mem_percent': u'25', u'is_container': u'false', u'redolog_size_in_mb': u'100 ', u'num_pdbs': u'1', u'pdb_prefix': u'pdb', u'oracle_db_name': u'racdb', u'oracle_edition': u'EE', u'oracle_database_type': u'MULTIPURPOSE', u'oracle_version_db': u'11.2.0.4'}]) => {"failed": true, "item": [{"filename": "p13390677_112040_Linux-x86-64_1of7.zip", "version": "11.2.0.4"}, {"home": "rachome1", "is_container": "false", "is_racone": "false", "num_pdbs": "1", "oracle_database_type": "MULTIPURPOSE", "oracle_db_mem_percent": "25", "oracle_db_name": "racdb", "oracle_db_passwd": "Password1", "oracle_db_type": "RAC", "oracle_edition": "EE", "oracle_init_params": "open_cursors=300,processes=500", "oracle_version_db": "11.2.0.4", "pdb_prefix": "pdb", "redolog_size_in_mb": "100 ", "service_name": "racdb_serv", "storage_type": "ASM"}]}
==> collabn1: msg: Source '/u01/stage/p13390677_112040_Linux-x86-64_1of7.zip' does not exist
==> collabn1: failed: [collabn1] => (item=[{u'version': u'11.2.0.4', u'filename': u'p13390677_112040_Linux-x86-64_2of7.zip'}, {u'oracle_init_params': u'open_cursors=300,processes=500', u'is_racone': u'false', u'oracle_db_type': u'RAC', u'service_name': u'racdb_serv', u'storage_type': u'ASM', u'oracle_db_passwd': u'Password1', u'home': u'rachome1', u'oracle_db_mem_percent': u'25', u'is_container': u'false', u'redolog_size_in_mb': u'100 ', u'num_pdbs': u'1', u'pdb_prefix': u'pdb', u'oracle_db_name': u'racdb', u'oracle_edition': u'EE', u'oracle_database_type': u'MULTIPURPOSE', u'oracle_version_db': u'11.2.0.4'}]) => {"failed": true, "item": [{"filename": "p13390677_112040_Linux-x86-64_2of7.zip", "version": "11.2.0.4"}, {"home": "rachome1", "is_container": "false", "is_racone": "false", "num_pdbs": "1", "oracle_database_type": "MULTIPURPOSE", "oracle_db_mem_percent": "25", "oracle_db_name": "racdb", "oracle_db_passwd": "Password1", "oracle_db_type": "RAC", "oracle_edition": "EE", "oracle_init_params": "open_cursors=300,processes=500", "oracle_version_db": "11.2.0.4", "pdb_prefix": "pdb", "redolog_size_in_mb": "100 ", "service_name": "racdb_serv", "storage_type": "ASM"}]}
==> collabn1: msg: Source '/u01/stage/p13390677_112040_Linux-x86-64_2of7.zip' does not exist
==> collabn1:

fixed with this code

git diff oraswdb-install/tasks/main.yml
diff --git a/oraswdb-install/tasks/main.yml b/oraswdb-install/tasks/main.yml
index 5c917bb..bed536f 100644
--- a/oraswdb-install/tasks/main.yml
+++ b/oraswdb-install/tasks/main.yml
@@ -103,7 +103,7 @@
       - oradbswunpack
- name: Extract files to stage-area (from remote location)
  -    unarchive: src={{ oracle_stage }}/{{ item[0].filename }}  dest={{ oracle_stage }}/{{ item[0].version }} copy=no
  -    unarchive: src={{ oracle_stage_remote }}/{{ item[0].filename }}  dest={{ oracle_stage }}/{{ item[0].version }} copy=no
    with_nested:
    - oracle_sw_image_db
      - oracle_databases
1. next time in next set when was trying to install oracle binaries.

/app/oracle/11.2.0.4/rachome1 /etc/oratab |wc -l'}}])
==> collabn1:
==> collabn1: TASK: [oradb-create | Create database(s)] *************************************
==> collabn1: fatal: [collabn2] => Traceback (most recent call last):
==> collabn1:   File "/usr/lib/python2.6/site-packages/ansible/runner/**init**.py", line 590, in _executor
==> collabn1:     exec_rc = self._executor_internal(host, new_stdin)
==> collabn1:   File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 818, in _executor_internal
==> collabn1:     complex_args=complex_args
==> collabn1:   File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 870, in _executor_internal_inner
==> collabn1:     if not utils.check_conditional(cond, self.basedir, inject, fail_on_undefined=self.error_on_undefined_vars):
==> collabn1:   File "/usr/lib/python2.6/site-packages/ansible/utils/__init__.py", line 274, in check_conditional
## ==> collabn1:     conditional = template.template(basedir, conditional, inject, fail_on_undefined=fail_on_undefined)

==> collabn1:   File "/usr/lib/python2.6/site-packages/ansible/utils/template.py", line 116, in template
==> collabn1:     varname = template_from_string(basedir, varname, templatevars, fail_on_undefined)
==> collabn1:   File "/usr/lib/python2.6/site-packages/ansible/utils/template.py", line 362, in template_from_string
==> collabn1:     res = jinja2.utils.concat(rf)
==> collabn1:   File "<template>", line 9, in root
==> collabn1:   File "/usr/lib64/python2.6/site-packages/jinja2/runtime.py", line 445, in _fail_with_undefined_error
==> collabn1:     raise self._undefined_exception(hint)
==> collabn1: UndefinedError: 'dict' object has no attribute 'state'

Fixed with this code change

git diff oradb-create/tasks/main.yml
diff --git a/oradb-create/tasks/main.yml b/oradb-create/tasks/main.yml
index 29e3fd0..9817a84 100644
--- a/oradb-create/tasks/main.yml
+++ b/oradb-create/tasks/main.yml
@@ -35,7 +35,7 @@
        - checkdbexist.results
     sudo: yes
     sudo_user: "{{ oracle_user }}"
-    when: master_node and item.1.stdout != "1" and "{{ item.0.state }}"|upper == "PRESENT"
-    when: master_node and item.1.stdout != "1"
   tags:
  - oradbcreate
    register: oradbcreate

I'm not use if this is related to my Mac or this is a generic issue.
I was installing 11.2.0.4 GI and DB.

regards,
Marcin

**Comment by oravirt** [comment]
I'll look into it.

/Micke

**Comment by oravirt** [comment]
Ok,

So in the last push I made to ansible-oracle (19/4) there was a copy paste
error (oracle_stage_remote got changed to oracle_stage) which I didn't
catch (shame on me). Fixed now though.

I'm not entirely sure what could have caused the other error though (at
least if you're installing 11.2.0.4). The 'state' property was added to the
'oracle_databases' list a while back, and the corresponding change was made
to the racattack playbook. Unfortunately, It wasn't added to the
'oracle_databases' list in racattack.group_vars though, but that should
only affect the default installation (12.1.0.2)

Anyway, these issues should have been resolved now.

regards
/Micke
